chain/rosetta-rpc/src/models.rs

```diff
@@ -948,12 +948,24 @@ pub(crate) struct Transaction {
 
     pub operations: Vec<Operation>,
 
-    /// Transactions that are related to other transactions (like a cross-shard
-    /// transaction) should include the transaction_identifier of these
-    /// transactions in the metadata.
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub related_transactions: Vec<RelatedTransaction>,
+
     pub metadata: TransactionMetadata,
 }
 
+/// Transaction related to another [`Transaction`] such as matching transfer or
+/// a cross-shard transaction.
+#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize, Apiv2Schema)]
+pub(crate) struct RelatedTransaction {
+    // Rosetta API defines network_identifier as well but since all our related
+    // transactions are always on the same network we can leave out this field.
+    // pub network_identifier: NetworkIdentifier,
+    pub transaction_identifier: TransactionIdentifier,
+
+    pub direction: String,
```

Comment:
I would use a custom enum instead of a String (see `SubAccount`, `CurrencySymbol`, `SyncStage` as examples).